spec.html

```diff
@@ -3789,17 +3807,17 @@ <h1>GetValue ( _V_ )</h1>
         <p>The abstract operation GetValue takes argument _V_. It performs the following steps when called:</p>
         <emu-alg>
           1. ReturnIfAbrupt(_V_).
-          1. If Type(_V_) is not Reference, return _V_.
-          1. Let _base_ be GetBase(_V_).
+          1. If _V_ is not a Reference Record, return _V_.
           1. If IsUnresolvableReference(_V_) is *true*, throw a *ReferenceError* exception.
+          1. Let _base_ be _V_.[[Base]].
           1. If IsPropertyReference(_V_) is *true*, then
             1. If HasPrimitiveBase(_V_) is *true*, then
-              1. Assert: In this case, _base_ will never be *undefined* or *null*.
-              1. [id="step-getvalue-toobject"] Set _base_ to ! ToObject(_base_).
-            1. Return ? _base_.[[Get]](GetReferencedName(_V_), GetThisValue(_V_)).
+              1. Assert: _base_ is an ECMAScript language value other than *undefined* or *null*.
+              1. [id="step-getvalue-toobject"] Let _baseObj_ be ! ToObject(_base_).
+            1. Return ? _baseObj_.[[Get]](_V_.[[ReferencedName]], GetThisValue(_V_)).
```

Comment:
Needs to assign something to _baseObj_ when HasPrimitiveBase is false, or unconditionally call ToObject(_base_).

Reply:
7a7a585 calls `ToObject` unconditionally, like in step **5.b** of [`delete` operator](https://tc39.es/ecma262/#sec-delete-operator-runtime-semantics-evaluation), and removes `HasPrimitiveBase` because:
* it doesn't seem justified to be in Reference's _public API_, unlike other `Is*Reference` methods;
* it doesn't feel right to query `base` type indirectly via an op.

`!` prefix asserts that `[[Base]]` value is not `undefined` or `null`.